src/VisualStudio/Core/Def/Implementation/AnalyzerDependency/AnalyzerFileWatcherService.cs

```diff
@@ -146,7 +143,7 @@ private void Tracker_UpdatedOnDisk(object sender, EventArgs e)
             var projectsWithAnalyzer = _workspace.ProjectTracker.Projects.Where(p => p.CurrentProjectAnalyzersContains(filePath)).ToArray();
             foreach (var project in projectsWithAnalyzer)
             {
-                RaiseAnalyzerChangedWarning(project.Id, filePath);
+                RaiseAnalyzerChangedWarningAsync(project.Id, filePath);
```

Comment:
this seems a bit dangerous too. like event fired but not registered to update source yet, but solution got closed. or even reopened.